keps/sig-storage/3314-csi-changed-block-tracking/README.md

```diff
@@ -0,0 +1,1646 @@
+used to simulate gRPC requests to the `SnapshotMetadata` service.
+
+The e2e tests will test the `SnapshotMetadata` service ability to:
+
+* Handle and authenticate gRPC requests from the sample client to:
+  * Get allocated blocks of a PVC
+  * Get changed blocks between a pair of snapshots
+* Get the following kinds of API resources, based on the gRPC requests and CSI
+provisioner:
+  * `SnapshotMetadataService`
+  * `TokenReview`
+  * `SubjectAccessReview`
+  * `VolumeSnapshot`
+  * `VolumeSnapshotContent`
+* Marshal and stream responses from the mock SP service to the sample client
+
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, [feature gate] graduations, or as
+something else. The KEP should keep this high-level with a focus on what
+signals will be looked at to determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Feature gate][feature gate] lifecycle
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+-->
+
+#### Alpha
+
+* Approval of the proposed CRDs and GRPC specification.
+* The `SnapshotMetadata` service can handle gRPC requests to get allocated and
+changed blocks, and stream responses back to client, without imposing load on
+the K8s API server.
+* Initial e2e tests completed and enabled.
+
+<!--
+#### Beta
+
+#### GA
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
```

Comment:
I would expect some idea of what's needed to get from alpha->beta and beta->GA too. In particular, I think something we'd want before beta is multiple implementations of both the SP side and the backup applicaiton side all interoperating without problems.

Reply:
I added some high-level criteria for beta and GA at aba16ff740090fe220b2519300a6722e44fa9c9a. PTAL - thanks!